src/app/src/main/kotlin/org/vaccineimpact/orderlyweb/WebEndpoint.kt

```diff
@@ -0,0 +1,73 @@
+package org.vaccineimpact.orderlyweb
+
+import org.vaccineimpact.orderlyweb.models.PermissionRequirement
+import org.vaccineimpact.orderlyweb.models.UserSource
+import org.vaccineimpact.orderlyweb.security.SkipOptionsMatcher
+import org.vaccineimpact.orderlyweb.security.WebSecurityConfigFactory
+import org.vaccineimpact.orderlyweb.security.authorization.OrderlyWebAuthorizer
+import org.vaccineimpact.orderlyweb.security.clients.MontaguIndirectClient
+import spark.Spark
+import spark.route.HttpMethod
+import kotlin.reflect.KClass
+
+data class WebEndpoint(
+        override val urlFragment: String,
+        override val controller: KClass<*>,
+        override val actionName: String,
+        override val method: HttpMethod = HttpMethod.get,
+        override val requiredPermissions: List<PermissionRequirement> = listOf(),
+        override val secure: Boolean = false
+) : EndpointDefinition
+{
+    override val transform = false
+    override val contentType = ContentTypes.html
+    override val allowParameterAuthentication = false
+    override val authenticateWithExternalProvider: Boolean = true
+
+    init
+    {
+        if (!urlFragment.endsWith("/"))
+        {
+            throw Exception("All endpoint definitions must end with a forward slash: $urlFragment")
+        }
+    }
+
+    override fun additionalSetup(url: String)
+    {
+        if (secure)
+        {
+            addSecurityFilter(url)
+        }
+    }
+
+    private fun addSecurityFilter(url: String)
```

Comment:
I was also assuming we'd hit API endpoints. I haven't thought about auth in depth, but I guess the easiest thing would be to include the `MontaguIndirectClient` to the API endpoints as well, so that if the user is logged in they can hit any endpoint.